symengine/sets.h

```diff
@@ -392,6 +399,26 @@ inline RCP<const Set> imageset(const RCP<const Basic> &sym,
         return finiteset({expr});
     }
 
+    if (is_a<FiniteSet>(*base)) {
+        map_basic_basic d;
+        set_basic temp;
```

Comment:
Name this something else. Use `[ci skip]` for next commit if these are the only changes you make.